contents/architecture/RESTArchitecturalStyle.md

```diff
@@ -0,0 +1,226 @@
+# REST Architectural Style
+
+Author: [Wen Xin](https://github.com/wenmogu)
+
+This book chapter introduces the REST architectural style, which is the guideline of development for the Web we use today. The following content is a summary of selected parts from the famous [Roy Fielding's paper about REST](https://www.ics.uci.edu/~fielding/pubs/dissertation/fielding_dissertation.pdf) with my own understanding, which I hope can serve as a relatively brief, easy-to-understand and at the same time detailed introduction to REST.
+
+**Requirement of Web development --- Background of REST architectural style**
+
+The forefather of Web Sir Tim Berners-Lee envisioned the Web to be &quot;a shared information space through which people and machines could communicate.&quot; Fielding&#39;s understanding of &quot;this shared information space&quot; is that the Web is a place where everyone&#39;s information is structured and stored in a way such that their own information &quot;could be usable to themselves and others&quot;, and they can &quot;reference and structure&quot; others&#39; information such that they do not need to keep a local copy of it.
+
+The design of the Web needs to consider the properties of its users and their information. The target audience of the Web are people in different organizations all over the world connected via the Internet. Their machines to access the Web might be very different, with different operating systems and requesting different file formats. These people&#39;s information also has a large variety in terms of the content type and the format. These restraints give rise to the following requirements of the Web:
+
+- Low entry barrier: the Web should have a simple, uniform interface for everyone to use
+- Extensibility: the Web service should be able to evolve to cater to the users&#39; evolving needs
+- Distributed Hypermedia: Hypermedia is a medium for information like graphics, audio, video, plain text and hyperlinks. The Web should use hypermedia as the user interface &quot;because of its simplicity and generality&quot;. Hypermedia has &quot;the same interface&quot; which can be used &quot;regardless of the information source&quot;, and can be used to structure information flexibly. Moreover, the hyperlinks embedded can guide the reader through the information. However, since hypermedia contains information like graphics, audio etc., the amount of information to be transferred is huge. Hence, the Web should be able to transfer a large amount of information within acceptable time limits.
```

Comment:
Are all the inline quotes from Fielding's paper? :/

Reply:
The point on distributed hypermedia can definitely be made clearer - there are actually only two main points (distributed and hypermedia) but as the paragraph grew haphazardly the main points have been obscured.